shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -0,0 +1,504 @@
+// Copyright 2018 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#define FML_USED_ON_EMBEDDER
+
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterEngine_Internal.h"
+
+#include <memory>
+
+#include "flutter/fml/message_loop.h"
+#include "flutter/fml/platform/darwin/platform_version.h"
+#include "flutter/shell/common/engine.h"
+#include "flutter/shell/common/platform_view.h"
+#include "flutter/shell/common/shell.h"
+#include "flutter/shell/common/switches.h"
+#include "flutter/shell/common/thread_host.h"
+#include "flutter/shell/platform/darwin/common/command_line.h"
+#include "flutter/shell/platform/darwin/ios/framework/Source/FlutterDartProject_Internal.h"
+#include "flutter/shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.h"
+#include "flutter/shell/platform/darwin/ios/framework/Source/FlutterTextInputDelegate.h"
+#include "flutter/shell/platform/darwin/ios/framework/Source/FlutterViewController_Internal.h"
+#include "flutter/shell/platform/darwin/ios/framework/Source/platform_message_response_darwin.h"
+#include "flutter/shell/platform/darwin/ios/platform_view_ios.h"
+
+@interface FlutterEngine () <FlutterTextInputDelegate>
+@property(nonatomic, readonly) NSMutableDictionary* pluginPublications;
```

Comment:
I know you just moved it but can we document this too? i.e. what is a publication vs registration vs registrar

Reply:
Done